packages/core-app-api/src/app/AppManager.tsx

```diff
@@ -154,7 +155,66 @@ function useConfigLoader(
     };
   }
 
-  const configReader = ConfigReader.fromConfigs(config.value ?? []);
+  let configReader;
+  /**
+   * config.value can be undefined or empty. If it's either, don't bother overriding anything.
+   */
+  if (config.value?.length) {
+    const urlConfigReader = ConfigReader.fromConfigs(config.value);
+
+    const resolveRelativeUrl = (relativeUrl: string) =>
+      new URL(relativeUrl, document.location.origin).href;
+
+    const getRelativeUrl = (fullUrl: string) => {
+      const url = new URL(fullUrl);
+      return fullUrl.replace(url.origin, '');
+    };
+
+    const getOrigin = (url: string) => new URL(url).origin;
+
+    const appBaseUrl = urlConfigReader.getOptionalString('app.baseUrl');
+    const backendBaseUrl = urlConfigReader.getOptionalString('backend.baseUrl');
+    let configs = config.value;
+    const relativeResolverConfig: { data: JsonObject; context: string } = {
+      data: {},
+      context: 'relative-resolver',
+    };
+    if (appBaseUrl && backendBaseUrl) {
+      const appOrigin = getOrigin(appBaseUrl);
+      const backendOrigin = getOrigin(backendBaseUrl);
+
+      /**
+       * We only want to override the URLs with the document origin when the URLs match
+       *  and are defined. We use getOptionalString here to not throw when the app.baseUrl
+       *  and backend.baseUrl are not defined. If they are defined but not well formatted URLs
+       *  the above getRelativeUrl() method will throw.
```

Comment:
Feeling this comment is a bit all over the place, commenting on code both above and below, probably best to move the various pieces into those locations instead.